scripts/release.sh

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+echo "enter release string according to semantic versioning (e.g. v1.2.3)."
+read -r INPUT
+if [[ ! "${INPUT}" =~ ^v[0-9]+.[0-9]+.[0-9]+ ]]; then
+  echo "Expected 'version' param of the form 'v<major-version>.<minor-version>.<patch-version>' but got '${INPUT}'"
+  exit 1
+fi
+
+VERSION=${INPUT#v}
+RELEASE_VERSION="${VERSION}"
+MINOR_VERSION=$(echo "${VERSION}" | cut -d. -f 1-2)
+
+REPOSITORY=${REPOSITORY:-"git@github.com:etcd-io/bbolt.git"}
+REMOTE="${REMOTE:-"origin"}"
+
+# ensuring the minor-version is identical.
+source_version=$(grep -E "\s+Version\s*=" ./version/version.go | sed -e "s/.*\"\(.*\)\".*/\1/g")
+if [[ "${source_version}" != "${RELEASE_VERSION}" ]]; then
+   source_minor_version=$(echo "${source_version}" | cut -d. -f 1-2)
+   if [[ "${source_minor_version}" != "${MINOR_VERSION}" ]]; then
+     echo "Wrong bbolt minor version in version.go. Expected ${MINOR_VERSION} but got ${source_minor_version}. Aborting."
+     exit 1
+   fi
+fi
+
+# creating a branch to bump 'version.go'.
+date_string=$(date +%Y%m%d)
+local_branch_name="version_${date_string}"
+local_branch_err=$(git checkout -b "${local_branch_name}" | grep -E "error|fatal" || true )
+if [[ -n "${local_branch_err}" ]]; then
+  echo "${local_branch_err}"
+fi
+
+# bump 'version.go'.
+echo "Updating version from '${source_version}' to '${RELEASE_VERSION}' in 'version.go'"
+sed -i "s/${source_version}/${RELEASE_VERSION}/g" ./version/version.go
+
+# push 'version.go' to remote.
+echo "committing 'version.go'"
+git add ./version/version.go
+git commit -s -m "Update version to ${VERSION}"
+git push -u -f "${REMOTE}" "${local_branch_name}"
```

Comment:
I suggest removing the force. If we're pushing to the etcd-io remote, it should not be a good idea to force it (it may even be good to fail if it fails to push).

Reply:
so iiuc, this script creates a branch in `origin`, to be used to create a PR against `upstream` 

I don't think the script should push directly to `etcd-io` repos, please correct me if I'm wrong

Reply:
We should remove creating the branch due to https://github.com/etcd-io/bbolt/pull/903/files#r1955614583